src/pre_process/m_initial_condition.fpp

```diff
@@ -283,6 +285,1204 @@ contains
         end do
 
     end subroutine s_perturb_surrounding_flow ! ----------------------------
+	
+	subroutine s_superposition_instability_wave() ! ------------------------------
+		real(kind(0d0)), dimension(5,0:m,0:n,0:p) :: wave,wave1,wave2,wave_tmp
+		real(kind(0d0)) :: tr,ti
+		real(kind(0d0)) :: Ly
+		integer :: i,j,k
+
+		Ly = y_domain%end - y_domain%beg
+		
+		write(*,*) "generate instability waves ..."
+
+		wave = 0d0
+		wave1 = 0d0
+		wave2 = 0d0
+		if (p .eq. 0) then
+			call s_instability_wave(2*pi*4.0/Ly,0d0,tr,ti,wave_tmp,0d0)
+			wave = wave + wave_tmp
+			call s_instability_wave(2*pi*2.0/Ly,0d0,tr,ti,wave_tmp,0d0)
+			wave = wave + wave_tmp
+			call s_instability_wave(2*pi*1.0/Ly,0d0,tr,ti,wave_tmp,0d0)
+			wave = wave + wave_tmp
+			wave = wave*0.05
+		else
+			call s_instability_wave(2*pi*4.0/Ly,0d0,tr,ti,wave_tmp,0d0)
+			wave1 = wave1 + wave_tmp
+			call s_instability_wave(2*pi*2.0/Ly,0d0,tr,ti,wave_tmp,0d0)
+			wave1 = wave1 + wave_tmp
+			call s_instability_wave(2*pi*1.0/Ly,0d0,tr,ti,wave_tmp,0d0)
+			wave1 = wave1 + wave_tmp
+			
+			call s_instability_wave(2*pi*4.0/Ly, 2*pi*4.0/Ly,tr,ti,wave_tmp,2*pi*0.8147d0)
+			wave2 = wave2 + wave_tmp
+			call s_instability_wave(2*pi*2.0/Ly, 2*pi*2.0/Ly,tr,ti,wave_tmp,2*pi*0.9058d0)
+			wave2 = wave2 + wave_tmp
+			call s_instability_wave(2*pi*1.0/Ly, 2*pi*1.0/Ly,tr,ti,wave_tmp,2*pi*0.0270d0)
+			wave2 = wave2 + wave_tmp
+			call s_instability_wave(2*pi*4.0/Ly,-2*pi*4.0/Ly,tr,ti,wave_tmp,2*pi*0.5324d0)
+			wave2 = wave2 + wave_tmp
+			call s_instability_wave(2*pi*2.0/Ly,-2*pi*2.0/Ly,tr,ti,wave_tmp,2*pi*0.2975d0)
+			wave2 = wave2 + wave_tmp
+			call s_instability_wave(2*pi*1.0/Ly,-2*pi*1.0/Ly,tr,ti,wave_tmp,2*pi*0.7634d0)
+			wave2 = wave2 + wave_tmp
+
+			wave = 0.05*wave1+0.15*wave2
+		end if
+
+        do k = 0, p
+        do j = 0, n
+        do i = 0, m
+			q_prim_vf(mom_idx%beg  )%sf(i,j,k) = q_prim_vf(mom_idx%beg  )%sf(i,j,k)+wave(2,i,j,k)
+			q_prim_vf(mom_idx%beg+1)%sf(i,j,k) = q_prim_vf(mom_idx%beg+1)%sf(i,j,k)+wave(3,i,j,k)
+			if (p .gt. 0) then
+			q_prim_vf(mom_idx%beg+2)%sf(i,j,k) = q_prim_vf(mom_idx%beg+2)%sf(i,j,k)+wave(4,i,j,k)
+			end if
+		end do
+		end do
+		end do
+
+	end subroutine s_superposition_instability_wave
+
+    subroutine s_instability_wave(alpha,beta,tr,ti,wave,shift)
```

Comment:
what does this do? what are the parameters? please add comments that describe them.

Reply:
Comments are added.